components/application-gateway/internal/proxy/proxy.go

```diff
@@ -57,6 +63,12 @@ func New(
 	csrfTokenStrategyFactory csrf.TokenStrategyFactory,
 	config Config,
 	configRepository proxyconfig.TargetConfigProvider) ProxyHandler {
+	var passportEnricher *passport.RequestEnricher
+	if config.AnnotatePassportHeaders {
```

Comment:
I would rather introduce some new abstraction here that will be responsible for enriching requests (similar to Authorization Strategy Factory and CSRF Token Strategy Factory).